### PR TITLE
Kernel timers

### PR DESCRIPTION
**Pull request by ergo720**
With this, we go further with LLE USB but we are still not quite there yet. However, this is still beneficial by itself and can be delivered independently, so here it is.

### CODE REVIEW

**Comment by RadWolfie** [comment]
Since you had reverted the LLE USB + add SDL submodule commit. I'm reviewing the changes toward timer system.

File: src/core/kernel/exports/EmuKrnlKi.h [hunk 1]
```diff
@@ -30,23 +30,176 @@
...
+                                                                             \
+    }                                                                        \
+}
 
-xboxkrnl::BOOLEAN KiInsertTreeTimer(
-	IN xboxkrnl::PKTIMER Timer,
-	IN xboxkrnl::LARGE_INTEGER Interval
-);
\ No newline at end of file
+#define KiWaitSatisfyAny(_Object_, _Thread_) {                               \
```
[the remark this answers is not in the document]
Comment (ergo720) [reply src/core/kernel/exports/EmuKrnlKi.h]: Don't know, if I remember right, @LukeUsher wrote these when he implemented ```KeWaitForMultipleObjects``` and ```KeWaitForSingleObject```, I merely moved them around. You'll have to ask him about these macros.

[the remark this answers is not in the document]
Comment (RadWolfie) [reply src/core/kernel/exports/EmuKrnlKi.h]: Okay, I'll dismiss it for now.

**Comment by LukeUsher** [comment]
As this is a large change, it requires significant testing, we don't want to accidentally cause any regressions.

Once @ergo720 is happy, and once our users are happy there are no (obvious) regressions, then we can merge this.

**Comment by ergo720** [comment]
I'm happy with this now, so, if there are no regressions, it can be merged.

**Pull request merged**